UnitsNet.Tests/CustomCode/MolarityTests.cs

```diff
@@ -30,29 +30,104 @@ public class MolarityTests : MolarityTestsBase
     {
         protected override double CentimolesPerLiterInOneMolesPerCubicMeter => 1e-1;
         protected override double DecimolesPerLiterInOneMolesPerCubicMeter => 1e-2;
-        protected override double MicromolesPerLiterInOneMolesPerCubicMeter => 1e3;
-        protected override double MillimolesPerLiterInOneMolesPerCubicMeter => 1;
         protected override double MolesPerLiterInOneMolesPerCubicMeter => 1e-3;
+        protected override double MillimolesPerLiterInOneMolesPerCubicMeter => 1;
+        protected override double MolesPerCubicMeterInOneMolesPerCubicMeter => 1;
+        protected override double MicromolesPerLiterInOneMolesPerCubicMeter => 1e3;
         protected override double NanomolesPerLiterInOneMolesPerCubicMeter => 1e6;
         protected override double PicomolesPerLiterInOneMolesPerCubicMeter => 1e9;
-        protected override double MolesPerCubicMeterInOneMolesPerCubicMeter => 1;
+
+        protected override double MolarInOneMolesPerCubicMeter => 1E-3;
+        protected override double MillimolarInOneMolesPerCubicMeter => 1;
+        protected override double MicromolarInOneMolesPerCubicMeter => 1E3;
+        protected override double NanomolarInOneMolesPerCubicMeter => 1E6;
+        protected override double PicomolarInOneMolesPerCubicMeter => 1E9;
+
+        private static double MolarMassHClInGramsPerMole = 36.46;
+        private static double MassOfSubstanceInGrams = 5;
+        private static double VolumeOfSolutionInLiters = 1.2;
+
+        private static double ExpectedMolarityMolesPerLiter = 0.1142805; // molarity = 5 / (1.2 * 36.46) = 0.114 mol/l = 0.114 M
+        private static double ExpectedConcentrationInKgPerCubicMeter = 4.16667;
+        
+        private static double MolarMassOfEthanolInGramsPerMole = 46.06844;
+        private static double DensityOfEthanolInKgPerCubicMeter = 789;
+        private static double VolumeConcentration_0_5M_Ethanol = 29.19419518377693;
 
         [Fact]
-        public void ExpectDensityConvertedToMolarityCorrectly()
+        public void ExpectMassConcentrationConvertedToMolarityCorrectly()
```

Comment:
Hm, should we remove the old tests? Isn't it good to test both Density and MassConcentration? If I read your PR description right, you say that MassConcentration does not replace Density and they should both continue to exist?

Reply:
No according to my colleagues those operations don't make sense- we assumed that @JonathanDavies626 had meant to express concentrations instead, but they weren't able to tell me what substances were used. 
Another indicator was the name of the variable used- (... **Mass** _molecularWeight_  ..)

Reply:
Aha, that is helpful to learn. I am not proficient in this domain of units. I guess I can agree to remove/change these methods if they truly are incorrect, without having to wait for v5 release first. 

Any chance you can comment on this @JonathanDavies626 whether you agree to this change or not?

Reply:
It's been some time since I wrote this. I'll take a look and get back to you this evening, if that's okay.

Reply:
Great! Absolutely, no problem.

> I guess I can agree to remove/change these methods if they truly are incorrect, without having to wait for v5 release first.

I'm going to change my mind on this one. We shouldn't take any shortcuts here even if it is convenient. If we decide to remove any methods, we should mark them as obsolete and remove them in v5.

Reply:
@lipchev is correct. I had intended to express concentrations in the molarity conversions. Example use case is converting blood glucose measurements from mg/dL to mmol/L.

Reply:
Thanks @JonathanDavies626 , much appreciated!